src/Parallel/ResourceInfo.hpp

```diff
@@ -765,6 +778,13 @@ void ResourceInfo<Metavariables>::build_singleton_map(
          "number of auto exclusive singletons to be allocated is "
              << alg::accumulate(auto_exclusive_singletons_on_each_node, 0_st));
 
+  // procs_to_ignore_ is now complete. Now construct procs_with_elements_
+  for (size_t i = 0; i < num_procs; i++) {
+    if (procs_to_ignore_.find(i) == procs_to_ignore_.end()) {
+      procs_with_elements_.insert(i);
```

Comment:
this is just the list of procs not ignored. I do not know if it is possible to exclude procs when migrating elements and creating new elements, or to insist that a particular proc have an element.

Reply:
Yes that is correct that it assumes every non-ignored proc has elements. Maybe I can rename the function to `nonignored_procs()` to avoid confusion?

For just picking an element, we could use the `is_zeroth_element()` function or something similar. Won't necessarily be the same element before/after an AMR regrid, but it is guaranteed to exist.

I don't think we error if there are more procs than elements, but I'm not sure (I've never tried). I'm not sure what the Z-curve does in that case.

Reply:
What is the motivation for this PR, then?  I thought it was to give us a way to run something on only one element, but it sounds like we can do that more simply with `is_zeroth_element`.

Reply:
Yeah it sounds like we can do it easier with `is_zeroth_element`. However, if you wanted to guarantee the element was on a specific proc, then you couldn't use `is_zeroth_element`.

Regardless, I think it might still be a good idea to have the `ResourceInfo` in the global cache. It is global info that all components can take advantage of if need be. And if we continue to add things to it in the future, the new info will already be readily available to all components to use.

Reply:
OK, let's add it.  I'm generally wary of adding stuff without an intended use case, but this is pretty small.